Source/storm/storm_svid.cpp

```diff
@@ -240,28 +237,27 @@ bool SVidPlayBegin(const char *filename, int flags)
 
 bool SVidPlayContinue()
 {
-	if (smk_palette_updated(SVidSMK) != 0) {
-		SDL_Color colors[256];
-		const unsigned char *paletteData = smk_get_palette(SVidSMK);
-
-		for (int i = 0; i < 256; i++) {
-			colors[i].r = paletteData[i * 3 + 0];
-			colors[i].g = paletteData[i * 3 + 1];
-			colors[i].b = paletteData[i * 3 + 2];
+	SDL_Color colors[256];
+	uint8_t *paletteData = SVidPaletteBuffer.get();
+	Smacker_GetPalette(SVidHandle, paletteData);
+
+	for (int i = 0; i < 256; i++) {
+		colors[i].r = paletteData[i * 3 + 0];
+		colors[i].g = paletteData[i * 3 + 1];
+		colors[i].b = paletteData[i * 3 + 2];
 #ifndef USE_SDL1
-			colors[i].a = SDL_ALPHA_OPAQUE;
+		colors[i].a = SDL_ALPHA_OPAQUE;
 #endif
 
-			orig_palette[i].r = paletteData[i * 3 + 0];
-			orig_palette[i].g = paletteData[i * 3 + 1];
-			orig_palette[i].b = paletteData[i * 3 + 2];
-		}
-		memcpy(logical_palette, orig_palette, sizeof(logical_palette));
+		orig_palette[i].r = paletteData[i * 3 + 0];
+		orig_palette[i].g = paletteData[i * 3 + 1];
+		orig_palette[i].b = paletteData[i * 3 + 2];
+	}
+	memcpy(logical_palette, orig_palette, sizeof(logical_palette));
 
-		if (SDLC_SetSurfaceAndPaletteColors(SVidSurface.get(), SVidPalette.get(), colors, 0, 256) <= -1) {
-			Log("{}", SDL_GetError());
-			return false;
-		}
+	if (SDLC_SetSurfaceAndPaletteColors(SVidSurface.get(), SVidPalette.get(), colors, 0, 256) <= -1) {

```

Comment:
doing this every frame may severely hurt performance on some devices

Reply:
I was wondering about that. It's easy enough to expose a Boolean from libsmackerdec whenever the palette changes.